darts/timeseries.py

```diff
@@ -3005,6 +3037,598 @@ def map(
+                    if re.sub(f"{sample_prefix}.*$", "", df_col) in comps_to_transform
+                ]
+
+                filter_df_non_transformed = [
+                    df_col
+                    for df_col in ts_df.columns.to_list()
+                    if re.sub(f"{sample_prefix}.*$", "", df_col)
+                    in comps_not_to_transform
+                ]
+            else:
+                filter_df_columns = [
+                    df_col
+                    for df_col in ts_df.columns.to_list()
+                    if df_col in comps_to_transform
+                ]
+                filter_df_non_transformed = [
+                    df_col
+                    for df_col in ts_df.columns.to_list()
+                    if df_col in comps_not_to_transform
+                ]
+
+            function_group_kwargs, function_kwargs = _get_kwargs(
+                transformation, _builtin_transforms
+            )
+
+            if isinstance(fn, str):  # built-in function
+                function_group = _builtin_transforms[fn][0].__name__
+                function_name = _builtin_transforms[fn][1]
+
+                new_columns.extend(
+                    [
+                        f"{fn}{transformation['window'] if 'window' in transformation else idx}_{comp_name}"
+                        for comp_name in comps_to_transform
+                    ]
+                )
+                # apply transformation to the selected components
+                resulting_transformations = pd.concat(
+                    [
+                        resulting_transformations,
+                        getattr(
+                            getattr(ts_df[filter_df_columns], function_group)(
+                                **function_group_kwargs
+                            ),
+                            function_name,
+                        )(**function_kwargs),
+                    ],
+                    axis=1,
+                )
+
+            elif isinstance(
+                transformation["function"], Callable
+            ):  # user provided function
+                new_columns.extend(
+                    [
+                        f"userFn{transformation['window'] if 'window' in transformation else idx}_{comp_name}"
+                        for comp_name in comps_to_transform
+                    ]
+                )
+                resulting_transformations = pd.concat(
+                    [
+                        resulting_transformations,
+                        ts_df[filter_df_columns]
+                        .rolling(**function_group_kwargs)
+                        .apply(fn, **function_kwargs),
+                    ],
+                    axis=1,
+                )
+
+            else:
+                raise_log(
+                    ValueError("The transformation function is not valid."), logger
+                )
+
+            # filter out non_transformed components
+            if keep_non_transformed:
```

Comment:
The logic here seems to have some issues. For instance I'd have expected all components to be kept with `keep_non_transformed` is set. Yet in the example below it keeps only one of the two:
![image](https://user-images.githubusercontent.com/4806678/200675761-e7219faa-4037-4d59-9192-f67e754ad8f3.png)

Reply:
Also in this example it seems the user provided function is the same as mean (note: I've set both transforms to apply on first component only) ?
![image](https://user-images.githubusercontent.com/4806678/200676044-a4e4766a-ea35-4862-9f17-b4d779c61e76.png)

Reply:
I set `keep_non_transformed` such that we return only the non_transformed components along the transformed ones and not all components. Wouldn't there be possible issues if we return the original components as well if say we might also have lagged data in a model? or am I just thinking too much :sweat_smile: